### PR TITLE
[NewUI] SendV2-#6: Customize Gas Modal UI

### PR DESCRIPTION
**Pull request by danjm**
customizegasmodal-mobile

customizegasmodal-laptop

### CODE REVIEW

File: ui/app/components/input-number.js [hunk 1]
```diff
@@ -1,56 +1,52 @@
...
-
   this.setValue = this.setValue.bind(this)
 }
 
 InputNumber.prototype.componentWillMount = function () {
-  const { initValue = 0 } = this.props
+  const { value = 0 } = this.props
 
-  this.setState({ value: initValue })
+  this.setState({ value: Number(value) })
```
Comment (sdtsui) [inline ui/app/components/input-number.js]: If we're leaving state in this component (which I think is okay), why do we need L33?

File: ui/app/components/input-number.js [hunk 2]
```diff
@@ -1,56 +1,52 @@
...
+  if (newValue >= min && newValue <= max) {
     onChange(newValue)
   }
 }
 
 InputNumber.prototype.render = function () {
-  const { unitLabel, step = 1, placeholder } = this.props
-  const { value } = this.state
+  const { unitLabel, step = 1, placeholder, value = 0 } = this.props
+  const valueAsNum = Number(value)
```
Comment (sdtsui) [inline ui/app/components/input-number.js]: See above: not sure why we need this conversion / if we need it, why it needs to be a separate line. Can just be in L39.    I like reusing state in L18 though.

File: ui/app/components/input-number.js [hunk 3]
```diff
@@ -1,56 +1,52 @@
...
     h('span.gas-tooltip-input-detail', {}, [unitLabel]),
     h('div.gas-tooltip-input-arrows', {}, [
       h('i.fa.fa-angle-up', {
-        onClick: () => this.setValue(value + step),
+        onClick: () => this.setValue(addCurrencies(valueAsNum, step)),
       }),
       h('i.fa.fa-angle-down', {
         style: { cursor: 'pointer' },
-        onClick: () => this.setValue(value - step),
+        onClick: () => this.setValue(addCurrencies(valueAsNum, step * -1)),
```
Comment (sdtsui) [inline ui/app/components/input-number.js]: Neat!

File: ui/app/components/send/gas-modal-card.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    onChange,
+    unitLabel,
+    value,
+    min,
+    max,
+    step,
+    title,
+    copy
+  } = this.props
+  console.log(`value`, value);
```
Comment (sdtsui) [inline ui/app/components/send/gas-modal-card.js]: stray log?

File: ui/app/components/modals/customize-gas-modal.js [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+      ]),
+
+      h('div.send-v2__customize-gas__body', {}, [
+        
+        h(GasModalCard, {
+          value: gasPrice,
+          min: 0.0,
+          max: 5.0,
+          step: 0.01,
+          onChange: value => this.setState({ gasPrice: value }),
```
Comment (sdtsui) [inline ui/app/components/modals/customize-gas-modal.js]: super minor: could be
`onChange: gasPrice => this.setState({ gasPrice }),`
same on L64

**Review by sdtsui — APPROVED** [review]
@danjm - I think you did a great job of breaking this into separate components. From what I can see, the hierarchy looks like so:
```
- Customize GasModal
-- GasModalCard
---- InputNumber
---- GasSlider
```

Might be best if you did this later, after the component is done, but have you considered moving this into its own directory?  I bring it up because it doesn't seem like these components will be reused anywhere. Again, I agree with how they were broken up though!  

This might be useful to keep the gas modal modular -- IIRC when the feature was originally designed, there was talk of making it reusable across different DApps.

File: ui/app/components/input-number.js (hunk 1 above)
Comment (danjm) [reply ui/app/components/input-number.js]: I meant to take state out of this component. Fixed in 32098a64

File: ui/app/components/input-number.js (hunk 2 above)
Comment (danjm) [reply ui/app/components/input-number.js]: Removed in 32098a64

File: ui/app/components/send/gas-modal-card.js (hunk 4 above)
Comment (danjm) [reply ui/app/components/send/gas-modal-card.js]: removed in fb35af8b

File: ui/app/components/modals/customize-gas-modal.js (hunk 5 above)
Comment (danjm) [reply ui/app/components/modals/customize-gas-modal.js]: change applied in fb35af8b

**Comment by danjm** [comment]
> Might be best if you did this later, after the component is done, but have you considered moving this into its own directory?

@sdtsui Great suggestion. Done in fb35af8b

**Comment by danjm** [comment]
changes merged with https://github.com/MetaMask/metamask-extension/pull/2364

**Pull request closed without merging**